network/network_test.go

```diff
@@ -0,0 +1,195 @@
+package network
+
+import (
+	"net"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestNetwork(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Network Suite")
+}
+
+var (
+	_ = Describe("Test Network", func() {
+
+		Describe("Test newExternalInterface", func() {
+			Context("When external interface alread exists", func() {
+				It("Should return nil", func() {
```

Comment:
have to check if nm.ExternalInterfaces is updated with "subnet" right?

Reply:
Yes, very good suggestion!